support-frontend/assets/pages/[countryGroupId]/components/checkoutComponent.tsx

```diff
@@ -429,244 +348,53 @@ export function CheckoutComponent({
-				promoCode !== undefined
-					? {
-							promoCode,
-							countryGroupId: geoId,
-					  }
-					: undefined;
-			const supportAbTests = getSupportAbTests(abParticipations);
-			const createSupportWorkersRequest: RegularPaymentRequest = {
-				...personalData,
-				billingAddress,
-				deliveryAddress,
-				firstDeliveryDate,
+			await submitForm({
+				geoId,
+				productKey,
+				ratePlanKey,
+				formData,
+				paymentMethod,
 				paymentFields,
-				appliedPromotion,
-				ophanIds,
-				referrerAcquisitionData,
-				product: productFields,
-				supportAbTests,
-				debugInfo: '',
-			};
-			setErrorMessage(undefined);
-			setErrorContext(undefined);
-
-			const createResponse = await fetch('/subscribe/create', {
-				method: 'POST',
-				body: JSON.stringify(createSupportWorkersRequest),
-				headers: {
-					'Content-Type': 'application/json',
-				},
+				productFields,
+				hasDeliveryAddress: !!productDescription.deliverableTo,
+				abParticipations,
+				promotion,
+				contributionAmount,
 			});
-			let processPaymentResponse: ProcessPaymentResponse;
-			let userType: UserType | undefined;
-
-			if (createResponse.ok) {
-				const statusResponse =
-					(await createResponse.json()) as CreateSubscriptionResponse;
-				userType = statusResponse.userType;
-				processPaymentResponse = await processPaymentWithRetries(
-					statusResponse,
-				);
-			} else {
-				const errorReason = (await createResponse.text()) as ErrorReason;
-				processPaymentResponse = {
-					status: 'failure',
-					failureReason: errorReason,
-				};
-			}
-
-			if (
-				processPaymentResponse.status === 'success' ||
-				processPaymentResponse.status === 'pending'
-			) {
-				const order = {
-					firstName: personalData.firstName,
-					email: personalData.email,
-					paymentMethod: paymentMethod,
-					status: processPaymentResponse.status,
-				};
-				setThankYouOrder(order);
-				const thankYouUrlSearchParams = new URLSearchParams();
-				thankYouUrlSearchParams.set('product', productKey);
-				thankYouUrlSearchParams.set('ratePlan', ratePlanKey);
-				promoCode && thankYouUrlSearchParams.set('promoCode', promoCode);
-				userType && thankYouUrlSearchParams.set('userType', userType);
-				contributionAmount &&
-					thankYouUrlSearchParams.set(
-						'contribution',
-						contributionAmount.toString(),
-					);
-				window.location.href = `/${geoId}/thank-you?${thankYouUrlSearchParams.toString()}`;
+		} catch (error) {
+			if (error instanceof FormSubmissionError) {
+				setErrorMessage(error.message);
+				setErrorContext(error.context);
 			} else {
-				console.error(
-					'processPaymentResponse error:',
-					processPaymentResponse.failureReason,
-				);
-				setErrorMessage('Sorry, something went wrong.');
-				setErrorContext(
-					appropriateErrorMessage(
-						processPaymentResponse.failureReason ?? 'unknown',
-					),
-				);
-				setIsProcessingPayment(false);
+				setErrorMessage('Sorry, something went wrong');
```

Comment:
I wonder whether we should log errors that aren't `FormSubmissionError` in Sentry? I don't think we'd want to do that for `FormSubmissionError` errors as they're known errors that could occur and are handled with meaningful messages displayed.